commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/CommerceProductUpgradeStepRegistrator.java

```diff
@@ -156,6 +170,9 @@ public void register(Registry registry) {
 	@Reference
 	private ConfigurationProvider _configurationProvider;
 
+	@Reference
+	private CPDefinitionLocalService _cpDefinitionLocalService;
```

Comment:
I think this is the problem. This service won't be registered until after the upgrade runs (because of the schema version requirement), but the upgrade won't run until this service is registered. Deadlock.

Reply:
i.e. so you have to do without this service, and run a query manually for the data you need.

Reply:
Oh shoot yeah :(

Reply:
@alecsloan looks like you removed usage of this service, but still left the reference in? I think that'll still prevent it from running properly.